### ISSUE TITLE
Failures on [cfe] Handle nnbd is-test in constant evaluation

### ISSUE THREAD

**Issue by johnniwinther** [dart-lang/sdk#41512]
There are new test failures on [[cfe] Handle nnbd is-test in constant evaluation](https://dart-review.googlesource.com/c/sdk/+/142982).

The tests
```
dart2js_extra/constant_truncate_test/e0 RuntimeError (expected Pass)
dart2js_extra/constant_truncate_test/e1 RuntimeError (expected Pass)
dart2js_extra/constant_truncate_test/e2 RuntimeError (expected Pass)
dart2js_extra/constant_truncate_test/f0 RuntimeError (expected Pass)
dart2js_extra/constant_truncate_test/f1 RuntimeError (expected Pass)
dart2js_extra/constant_truncate_test/f2 RuntimeError (expected Pass)
dart2js_extra/constant_truncate_test/none RuntimeError (expected Pass)
```
are failing on configurations
```
dart2js-hostasserts-strong-linux-x64-d8
```

**Comment by johnniwinther** [comment]
The `dart2js_extra/constant_folding2_test` test also failed on this commit (flakyly?)

**Comment by fishythefish** [comment]
Should be fixed by 1ac89cd5b3ed6dde5830fc3c3b8c14a07ee79916.
